### ISSUE TITLE
Feature Request: GCode Dry Run

### ISSUE THREAD

**Issue by ajdale** [foosel/OctoPrint#487]
Adding the ability to do a 'dry run' of the GCode files without having the printer actually execute the instructions would be handy. Mainly to detect errors in the GCode file before the print gets underway.

I have had a file that I personally made turn up with an error where my print stops. It would be nice to know that the job will fail before starting it and walking away.

**Comment by foosel** [comment]
Apart from what is already done via the virtual printer (as outlined above), there's nothing more that OctoPrint can do here (such as a full blown GCODE evaluation). Nothing against people contributing plugin's though.

**Comment by rewolff** [comment]
@foosel, A friend wants his printer to show the motions "in real life" without actually heating the extruder and extruding plastic. Apparently he's worried about the printer going "crazy fast", which the rest of us would consider a good thing. :-) 

Anyway, five minutes later I have the core of the code there, and I'm investigating how to write a plugin....

**Comment by OllisGit** [comment]
@rewolff  check out this plugin: https://github.com/OllisGit/OctoPrint-DryRun
